### ISSUE TITLE
collection_methods_unrelated_type false positives for `intersection`

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/linter#4336]
**Describe the issue**

Any invocation of `.intersection` seems to always trigger this lint. The only way I can figure out how to use this function without triggering the lint is if the receiver is a `Set<Object?>`.

**To Reproduce**

```dart
void main() {
  var x = {'a'};
  print(x.intersection({'b'}); // The argument type 'Set<Object?>' isn't related to 'Set<String>'
}
```

**Expected behavior**

No lint :).

**Additional context**

This seems to be because the parameter of `.intersection` is typed `Set<Object?>`.

**Comment by srawlins** [comment]
This rule no longer reports `intersection`, as of https://github.com/dart-lang/linter/commit/286abec5ba51330d27777b574bd97e9570db251a. Please upgrade to a more recent version of Dart, or wait for the next release.
